### PR TITLE
further updates to the fulltext rewrite

### PR DESCRIPTION
**Pull request by JosuaKrause**
fulltext headers only on content

ran script on staging

### CODE REVIEW

File: public/js/contribute/pad/save.js [hunk 1]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
     datum.has_content = datum.txt?.trim()?.length > 0;
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
```
Comment (myjyby) [inline public/js/contribute/pad/save.js:94]: what does the innerText var do more than datum.has_content here? (and above)

File: public/js/contribute/pad/save.js [hunk 2]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
...
+        .filter((b) => b.name?.length && b.checked)
+        .map((b) => b.name)
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${clone.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
```
Comment (myjyby) [inline public/js/contribute/pad/save.js:151]: again, this seems quite a lengthy rewrite to (if I get it correctly) simply check for 0 length strings. what am I missing?

File: public/js/contribute/pad/save.js [hunk 3]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
...
+      innerText += (datum.sdgs || datum.tags)
+        .map((b) => `${b.type}: ${b.name}`)
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
```
Comment (myjyby) [inline public/js/contribute/pad/save.js:184]: Same here: why not just use datum.has_content?

File: public/js/contribute/pad/save.js [hunk 4]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
...
+          return `${front}${b}`;
+        })
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
```
Comment (myjyby) [inline public/js/contribute/pad/save.js:206]: and again

File: routes/scripts/shared/fix_fullTxt.js [hunk 5]
```diff
@@ -11,29 +12,42 @@ DB.conn.tx(t => {
...
+            }
           }
         })
-      })
 
+        if (currentTxt.length) {
+          if (store_instructions && c.title) {
+            fullTxt += `${c.title}\n`;
+          }
+          if (store_instructions && c.lead) {
```
Comment (myjyby) [inline routes/scripts/shared/fix_fullTxt.js:42]: hmm this is not only for templates so I would remove the store_instuctions conditions here

File: public/js/contribute/pad/save.js (hunk 1 above)
Comment (JosuaKrause) [reply public/js/contribute/pad/save.js:94]: datum.txt.trim() can still be empty

Comment (JosuaKrause) [reply public/js/contribute/pad/save.js:94]: yes, you're right datum.txt cannot be empty here

I have the innerText changes everywhere in the same way to be consistent and make it easier to see whether the changes are the same everywhere

File: public/js/contribute/pad/save.js (hunk 2 above)
Comment (JosuaKrause) [reply public/js/contribute/pad/save.js:151]: innerText is needed to delay the update. innerText can be completely made of white space

File: public/js/contribute/pad/save.js (hunk 3 above)
Comment (JosuaKrause) [reply public/js/contribute/pad/save.js:184]: having the arrays not empty doesn't guarantee that the text won't be empty. that said, do we want both sdgs and tags here? that seems better. right now we show the tags only if sdgs is empty

File: routes/scripts/shared/fix_fullTxt.js (hunk 5 above)
Comment (JosuaKrause) [reply routes/scripts/shared/fix_fullTxt.js:42]: not sure what you mean. that's your code :)

File: public/js/contribute/pad/save.js [hunk 6]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
...
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
-    }
+    let innerText = '';
     if (datum.has_content) {
-      text += (sel.select('.media-embed').node() ||
+      const newText = (sel.select('.media-embed').node() ||
         sel.select('.meta-embed').node())[
         'outerText' || 'textContent' || 'innerText'
```
Comment (JosuaKrause) [inline public/js/contribute/pad/save.js:117]: check which one exists

File: routes/scripts/shared/fix_fullTxt.js [hunk 7]
```diff
@@ -49,33 +63,67 @@ function retrieveText(d) {
...
+      if (store_instructions && d.instruction) text += `${d.instruction}\n`;
+      text += `${innerText}\n`;
+    }
   } else if (d.type === 'attachment') {
-    if (store_instructions && d.instruction) text += `${d.instruction}\n`;
-    if (d.has_content) text += d.srcs.map((a) => `${a.name}: ${a}`).join('\n');
-  } 
-  return `${text}\n`
+    let innerText = '';
+    if (d.has_content) innerText += (d.srcs ?? []).map((a) => d.name ? `${d.name}: ${a}` : `${a}`).join('\n');
```
Comment (JosuaKrause) [inline routes/scripts/shared/fix_fullTxt.js:116]: no need to check whether d.name exists

File: public/js/contribute/pad/save.js [hunk 8]
```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
...
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
+    if (datum.has_content)
+      innerText += datum.srcs
+        .map((b) => {
+          const front = datum.name ? `${datum.name}: ` : '';
```
Comment (JosuaKrause) [inline public/js/contribute/pad/save.js:196]: not necessary

File: public/js/contribute/pad/save.js (hunk 6 above)
Comment (JosuaKrause) [reply public/js/contribute/pad/save.js:117]: three times in the file

File: routes/scripts/shared/fix_fullTxt.js (hunk 5 above)
Comment (JosuaKrause) [reply routes/scripts/shared/fix_fullTxt.js:42]: I just moved it from here https://github.com/UNDP-Accelerator-Labs/platform/pull/285/files#diff-e38c85e90175768e57039d33c465662d62c108dc4749bf4298a86918c31960a1L18

**Review by myjyby — APPROVED** [review]
lgtm

**Pull request merged**